frb_codegen/src/generator/rust/mod.rs

```diff
@@ -144,9 +158,11 @@ impl Generator {
 
         lines.push(self.section_header_comment("executor"));
         lines.push(self.generate_executor(ir_file));
-
-        lines.push(self.section_header_comment("sync execution mode utility"));
-        lines.push(self.generate_sync_execution_mode_utility());
+        
+        if self.options.should_generate_sync_execution_mode_utility.unwrap_or(true) {
```

Comment:
By default, we always want to generate the `free_WireSyncReturnStruct` function.